triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunner.scala

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.engine.trigger
+
+import akka.actor.typed.{Behavior, PostStop}
+import akka.actor.typed.scaladsl.AbstractBehavior
+import akka.actor.typed.SupervisorStrategy._
+import akka.actor.typed.Signal
+import akka.actor.typed.PostStop
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.ActorContext
+import akka.stream.Materializer
+import com.typesafe.scalalogging.StrictLogging
+import com.daml.grpc.adapter.ExecutionSequencerFactory
+
+object TriggerRunner {
+  type Config = TriggerRunnerImpl.Config
+
+  trait Message
+  final case object Stop extends Message
+
+  def apply(config: Config, name: String)(
+      implicit esf: ExecutionSequencerFactory,
+      mat: Materializer): Behavior[TriggerRunner.Message] =
+    Behaviors.setup(ctx => new TriggerRunner(ctx, config, name))
+}
+
+class TriggerRunner(
+    ctx: ActorContext[TriggerRunner.Message],
+    config: TriggerRunner.Config,
+    name: String)(implicit esf: ExecutionSequencerFactory, mat: Materializer)
+    extends AbstractBehavior[TriggerRunner.Message](ctx)
+    with StrictLogging {
+
+  import TriggerRunner.{Message, Stop}
+
+  private val child =
+    ctx.spawn(Behaviors.supervise(TriggerRunnerImpl(config)).onFailure(restart), name)
```

Comment:
Is there an analogue of this in `TriggerActor` or is this brand new?

Reply:
Re-reading your PR description, it looks like this entire class is new and this is the key line which does restarts on failure.

Reply:
`TriggerRunnerImpl` is basically that which was `TriggerActor`.